pkg/engine/artifacts.go

```diff
@@ -231,7 +231,7 @@ func kubernetesAddonSettingsInit(p *api.Properties) []kubernetesComponentFileSpe
 				sourceFile:      unmanagedStorageClassesSourceYaml,
 				base64Data:      p.OrchestratorProfile.KubernetesConfig.GetAddonScript(AzureStorageClassesAddonName),
 				destinationFile: "azure-storage-classes.yaml",
-				isEnabled:       p.AgentPoolProfiles[0].StorageProfile != api.ManagedDisks,
+				isEnabled:       p.AgentPoolProfiles[0].StorageProfile == api.StorageAccount,
```

Comment:
This is a much better implementation, as it matches the requirements of the `StorageProfile` only accepting either of these two values.